src/code.sh

```
@@ -1,29 +1,38 @@
+        exit 1
+    fi
 
-    # or files can be downloaded one by one, specifying a name for them within the workstation
-    dx download "$input_file" -o input_file_name
+    if [[ "$READ_COUNT" -eq 0 ]]; then
+        echo "ERROR: samtools flagstat reports zero primary reads in input BAM. Cannot calculate downsampling fraction." >&2
+        exit 1
+    fi
 
-    
+    TARGET_FRACTION=$(bc -l <<< "scale=3; $target_read_count / $READ_COUNT")
```

Shellcheck (0.11.0)

[warning] target_read_count is referenced but not assigned.

(SC2154)